### ISSUE TITLE
enable-proposed-api in argv.json doesn't work in Stable

### ISSUE THREAD

**Issue by usernamehw** [microsoft/vscode#106427]
#99775

Trying to use proposed api in Stable. Insiders channel works without problems:

```js
"enable-proposed-api": [
	"EXTENSION_ID"
],
```

But Stable version does not respect `argv.json` **enable-proposed-api** property and fails to activate the extension.

**Comment by gjsjohnmurray** [comment]
https://code.visualstudio.com/api/advanced-topics/using-proposed-api seems pretty clear that you have to be using Insiders if you want to use proposed APIs.

**Comment by JacksonKearl** [comment]
Hm. I'd expect the two to be identical (https://github.com/microsoft/vscode/pull/99908/files#diff-1c90ff38a08209f9ebd4d05d1e43358eR239), but I am able to repro.

**Comment by JacksonKearl** [comment]
@usernamehw do you know if this ever worked? If this is a regression we should fix it but I'm not 100% sure that we should otherwise... we make proposed api's a pain to use for a reason 🙃

**Comment by gjsjohnmurray** [comment]
@JacksonKearl so how long has that documentation been incorrect about this?

**Comment by JacksonKearl** [comment]
Those docs are referring to extension development, which does not make use of the CLI flag. Running `code --help` or attempting to use a proposed API extension documents the `--enable-proposed-api` flag, and I don't think we need to explicitly call out that it can be passed either via CLI or `argv.json` (that's the whole point of `argv.json`), but not sure.

**Comment by usernamehw** [comment]
> @usernamehw do you know if this ever worked?

Never tried it before.

> If this is a regression we should fix it but I'm not 100% sure that we should otherwise... we make proposed api's a pain to use for a reason 🙃

I would expect it to work. Proposed api should not be a pain, but rather a choice of someone to test newer api.

Maybe @bpasero knows why it doesn't work?

**Comment by bpasero** [comment]
As far as I recall, proposed API is only supposed to work in insiders. @jrieken or @alexdima to confirm.

**Comment by gjsjohnmurray** [comment]
> 
> 
> As far as I recall, proposed API is only supposed to work in insiders.

That has always been my understanding, and certainly what the doc at https://code.visualstudio.com/api/advanced-topics/using-proposed-api clearly states IMO.

**Comment by jrieken** [comment]
Yes - we don't allow proposed API in stable builds and have no plans to do so

**Comment by usernamehw** [comment]
@jrieken But it does work in Stable, at least for year or more. It's just fails from `argv.json`.

**Comment by JacksonKearl** [comment]
@jrieken @bpasero you are able to launch with proposed api via `code --enable-proposed-api {ID}`, is this expected?

**Comment by jrieken** [comment]
> @jrieken @bpasero you are able to launch with proposed api in stable via code --enable-proposed-api {ID}, is this expected?

I would say no... The point of proposed API is that it changes a lot and therefore only insider usage makes sense. You need to be on top of things and you cannot when using stable. Tho, I don't know why it is working with the flag - bug or feature?

**Comment by usernamehw** [comment]
I don't get it... How do you feed proposed api to users?

When using combo Insiders + proposed api - that means the extension can break every single update of vscode in turn breaking the extension and you need to update the extension very often? It also means forcing users to use Insiders version which breaks very often.

On the other hand using Stable version + proposed api - means you can test the extension once a month before Stable release and know that it will still work.

**Comment by jrieken** [comment]
We do have proposed API so that extension authors can provide us with feedback. That extension authors can build new stuff for extension users isn't our main priority here. If you only adopt proposed API changes only once a month and with the mindset of getting your feature fixed, then you likely won't be good at providing that feedback - which defeats the purpose of allowing use proposed API in the first place.

**Comment by jrieken** [comment]
Tho, having said that. If we do allow proposed API in stable via the explicit flag on purpose then we should also allow in the argv-file. However, I think it's questionable that we do for the reasons outlined above.

**Comment by bpasero** [comment]
All that `argv.json` seems to do is this: `process.argv.push('--enable-proposed-api', id)` so I do not see how this would behave any different from putting this on the command line. I do not understand how this could have ever worked in stable.

**Comment by bpasero** [comment]
Relevant code seems to be here:

https://github.com/microsoft/vscode/blob/df5d0e588ff6956c812b617895e4598ade4db428/src/vs/workbench/services/extensions/common/abstractExtensionService.ts#L479-L486

If I understand correctly, we only do not allow stable to develop on an extension with proposed API.

If there is a bug here then maybe something broke with regards to pushing args into the `process` in `main.js` and the renderer not being able to read it out from there.

@JacksonKearl maybe something to investigate and then reopen this issue for consideration.

**Comment by usernamehw** [comment]
Is issue accepted to  be a `bug` or `as-designed`. It's closed now and on the way to be locked.

**Comment by JacksonKearl** [comment]
Moving to `under-discussion`. On one hand I see the desire to allow for more dogfooding, especially as some proposed API's stay proposed for a very long time... on the other it's too easy for an extension readme to say "before using, go and add this line to your argv.json", not be updated in forever, and when the API inevitably changes a bunch of users have a broken experience. 

Considering enabling it for Stable but having some sort of scary message logged to console on startup for any entries in `argv.json/proposedApi`. Of course, not everybody (almost nobody) is going to open dev tools before filing an issue against us, so this still has the potential to create a lot of noise.

**Comment by JacksonKearl** [comment]
Spent some time digging into why this would be different with little to show for it -- I don't believe there's any logic explicitly checking for Insiders vs Stable so perhaps a timing bug?

Opening this up to the community if anyone is particularly passionate.

**Comment by usernamehw** [comment]
It's not possible to publish proposed api extension to Marketplace, so I don't actually see it working smoothly for users.

Instead, would it make sense to move simple proposals faster? For instance, this issue #32813 was implemented in December 2019, and had almost no changes, but it's still in a proposed state. Kind of feels like it's only moved to stable when you ping the assingee a magic number of times to annoy them enough 😅.
If it's not ready, it would be great to know what blocks it, or to know an approximate date.

**Comment by JacksonKearl** [comment]
And now I'm for some reason unable to use proposed entries in argv.json it even in Insiders..
